tls/s2n_client_finished.c

```diff
@@ -26,11 +26,13 @@
 #include "utils/s2n_safety.h"
 
 S2N_RESULT s2n_finished_recv(struct s2n_connection *conn, uint8_t *our_version);
-S2N_RESULT s2n_finished_send(struct s2n_connection *conn, uint8_t *seq_num, uint8_t *our_version);
+S2N_RESULT s2n_finished_send(struct s2n_connection *conn, uint8_t *our_version);
 
 int s2n_client_finished_recv(struct s2n_connection *conn)
 {
     POSIX_ENSURE_REF(conn);
+
+    POSIX_GUARD_RESULT(s2n_prf_client_finished(conn));
```

Comment:
As an example of the comment that s2n_prf_client_finished definitely needs: we do this on the CCS message usually because the hash shouldn't include the CCS message. But if you're doing npn, the hash needs to suddenly include the npn message AND the ccs message, so has to happen on the finished message.